postreise/plot/plot_curtailment_ts.py

```diff
@@ -0,0 +1,201 @@
+    get_generation_time_series_by_resources,
+)
+from postreise.analyze.time import (
+    change_time_zone,
+    resample_time_series,
+    slice_time_series,
+)
+
+
+def plot_curtailment_time_series(
+    scenario,
+    area,
+    resources,
+    area_type=None,
+    time_range=None,
+    time_zone="utc",
+    time_freq="H",
+    show_demand=True,
+    percentage=True,
+    t2c=None,
+    t2l=None,
+    title=None,
+    label_fontsize=20,
+    title_fontsize=22,
+    tick_fontsize=15,
+    legend_fontsize=18,
+    save=False,
+    filename=None,
+    filepath=None,
+):
+    """Generate time series curtailment plot of each specified resource in a certain
+    area of a scenario.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of: *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources.
+    :param str area_type: one of: *'loadzone'*, *'state'*,
+        *'state_abbr'*, *'interconnect'*
+    :param tuple time_range: (start time stamp, end time stamp). If None, the entire
+        time range is used for the given scenario.
+    :param str time_zone: new time zone.
+    :param str time_freq: frequency. Either *'D'* (day), *'W'* (week), *'M'* (month).
+    :param bool show_demand: show demand line in the plot or not, default is True.
+    :param bool percentage: plot the curtailment in terms of percentage or not,
+        default is True.
+    :param dict t2c: user specified color of resource type to overwrite type2color
+        default dict. key: resource type, value: color code.
+    :param dict t2l: user specified label of resource type to overwrite type2label
+        default dict. key: resource type, value: label.
+    :param str title: user specified title of the figure.
+    :param float label_fontsize: user specified label fontsize, default is 20.
+    :param float title_fontsize: user specified title fontsize, default is 22.
+    :param float tick_fontsize: user specified ticks of axes fontsize, default is 15.
+    :param float legend_fontsize: user specified legend fontsize, default is 18.
+    :param bool save: save the generated figure or not, default is False.
+    :param str filename: if save is True, user specified filename, use area if None.
+    :param str filepath: if save is True, user specified filepath, use current
+        directory if None.
+    """
+    _check_scenario_is_in_analyze_state(scenario)
+    resources = _check_resources_and_format(resources)
+
+    if t2c:
+        type2color.update(t2c)
+    if t2l:
+        type2label.update(t2l)
+
+    pg = get_generation_time_series_by_resources(
+        scenario, area, resources, area_type=area_type
+    )
+    demand = get_demand_time_series(scenario, area, area_type=area_type)
+    curtailment = get_curtailment_time_series(scenario, area, area_type=area_type)
+
+    if time_zone != "utc":
```

Comment:
What if the user specifies "UTC" (this is the name of the index of PG currently)? I think the logic of "do we need to actually change the timezone or are they equivalent" is probably better suited to live within `change_time_zone`. In there, we could do something like `if pytz.timezone(time_zone) != pytz.timezone(df.index.name)` (since the timezones of "UTC" and "utc" evaluate to be equal in this way).

Reply:
Actually, here we are just checking whether the user specified a time zone other than our default value "utc". It won't hurt if we remove this `if` statement and everything still works. In fact, we did `ts.tz_localize("UTC").tz_convert(time_zone) `in `change_time_zone`, and it can handle anything like "UTC", "Utc", "uTc", "uTC", ...